extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -697,37 +684,48 @@ public String getErrorMessage() {
     private static final String HTTPS = "https:";
     private static final String DEOBFUSCATION_FUNC_NAME = "deobfuscate";
 
-    private final static String[] REGEXES = {
+    private static final String[] REGEXES = {
             "(?:\\b|[^a-zA-Z0-9$])([a-zA-Z0-9$]{2})\\s*=\\s*function\\(\\s*a\\s*\\)\\s*\\{\\s*a\\s*=\\s*a\\.split\\(\\s*\"\"\\s*\\)",
             "([\\w$]+)\\s*=\\s*function\\((\\w+)\\)\\{\\s*\\2=\\s*\\2\\.split\\(\"\"\\)\\s*;",
             "\\b([\\w$]{2})\\s*=\\s*function\\((\\w+)\\)\\{\\s*\\2=\\s*\\2\\.split\\(\"\"\\)\\s*;",
-            "yt\\.akamaized\\.net/\\)\\s*\\|\\|\\s*.*?\\s*c\\s*&&\\s*d\\.set\\([^,]+\\s*,\\s*(:encodeURIComponent\\s*\\()([a-zA-Z0-9$]+)\\(",
             "\\bc\\s*&&\\s*d\\.set\\([^,]+\\s*,\\s*(:encodeURIComponent\\s*\\()([a-zA-Z0-9$]+)\\("
     };
+    private static final String STS_REGEX = "signatureTimestamp[=:](\\d+)";
 
     @Override
```

Comment:
@TiA4f8R I can't find anything related to your problem here

Reply:
@XiangRongLin With @mhmdanas, we already found a solution for this; now the real problem is that if the extractor fallbacks to the desktop endpoint, the JS player is fetched at every time and not cached, even if I store it (I also do it for the `signatureTimestamp`), in global objets of the app.
When I ran the unavailable tests of the `YoutubeStreamExtractorDefaultTest` class, the player code, the `signatureTimestamp` and the debofuscation code are null at each time a test of the unavailable class is ran. It wasn't the case before my changes.